telegram/ext/filters.py

```diff
@@ -372,1895 +363,1807 @@ def name(self, name: str) -> NoReturn:
-                * Pass :obj:`None` to filter files with no extension,
-                  i.e. without a dot in the filename.
-
-            Example:
-                * ``Filters.document.file_extension("jpg")``
-                  filters files with extension ``".jpg"``.
-                * ``Filters.document.file_extension(".jpg")``
-                  filters files with extension ``"..jpg"``.
-                * ``Filters.document.file_extension("Dockerfile", case_sensitive=True)``
-                  filters files with extension ``".Dockerfile"`` minding the case.
-                * ``Filters.document.file_extension(None)``
-                  filters files without a dot in the filename.
+        Use these filters like: ``filters.Document.MP3``,
+        ``filters.Document.MimeType("text/plain")`` etc. Or use just
+        ``filters.DOCUMENT`` for all document messages.
     """
 
-    class _Animation(MessageFilter):
-        __slots__ = ()
-        name = 'Filters.animation'
+    __slots__ = ()
+    name = 'filters.DOCUMENT'
 
-        def filter(self, message: Message) -> bool:
-            return bool(message.animation)
+    class Category(MessageFilter):
+        """Filters documents by their category in the mime-type attribute.
 
-    animation = _Animation()
-    """Messages that contain :class:`telegram.Animation`."""
+        Args:
+            category (:obj:`str`): Category of the media you want to filter.
 
-    class _Photo(MessageFilter):
-        __slots__ = ()
-        name = 'Filters.photo'
+        Example:
+            ``filters.Document.Category('audio/')`` returns :obj:`True` for all types
+            of audio sent as a file, for example ``'audio/mpeg'`` or ``'audio/x-wav'``.
 
-        def filter(self, message: Message) -> bool:
-            return bool(message.photo)
+        Note:
+            This Filter only filters by the mime_type of the document, it doesn't check the
+            validity of the document. The user can manipulate the mime-type of a message and
+            send media with wrong types that don't fit to this handler.
+        """
 
-    photo = _Photo()
-    """Messages that contain :class:`telegram.PhotoSize`."""
+        __slots__ = ('_category',)
 
-    class _Sticker(MessageFilter):
-        __slots__ = ()
-        name = 'Filters.sticker'
+        def __init__(self, category: str):
+            self._category = category
+            self.name = f"filters.Document.Category('{self._category}')"
 
         def filter(self, message: Message) -> bool:
-            return bool(message.sticker)
-
-    sticker = _Sticker()
-    """Messages that contain :class:`telegram.Sticker`."""
+            if message.document:
+                return message.document.mime_type.startswith(self._category)
+            return False
 
-    class _Video(MessageFilter):
-        __slots__ = ()
-        name = 'Filters.video'
+    APPLICATION = Category('application/')
+    """Use as ``filters.Document.APPLICATION``."""
```

Comment:
actually the docs now automatically show that. 
![image](https://user-images.githubusercontent.com/37377066/139918873-6c765d6d-b094-4825-a91b-0676a5fc38cc.png)

So I thought might as well show how to use it since that isn't obvious

Reply:
Ah cool! didn't build the docs :D then let's leave it this way 👍